src/lib/hash/sha3/sha3.h

```diff
@@ -10,14 +11,16 @@
 
 #include <botan/hash.h>
 #include <botan/secmem.h>
+#include <botan/internal/keccak_fips.h>
 #include <string>
 
 namespace Botan {
 
 /**
 * SHA-3
 */
-class SHA_3 : public HashFunction
+
+class SHA_3 : public Keccak_FIPS
```

Comment:
Inheritance here seems just wrong.

TBH I do not understand at all the motivation for moving the SHA-3 permutation logic, nor for how Keccak_FIPS simplifies anything.